charmbits/mongodbrelation/requirer.go

```diff
@@ -53,3 +77,45 @@ func unitAddress(vals map[string]string) (string, error) {
 	}
 	return net.JoinHostPort(host, port), nil
 }
+
+func init() {
+	httpservice.RegisterRelationType(
+		registerField,
+		setFieldValue,
+	)
+}
+
+type relationInfo struct {
+	URL string
+}
+
+func registerField(r *hook.Registry, tag string) func() (interface{}, error) {
+	var req Requirer
+	f := strings.Split(tag, ",")
+	relationName := "mongodb"
+	if f[0] != "" {
+		relationName = f[0]
+	}
+	// get "required" value from tag too?
+	req.Register(r, relationName)
+	return func() (interface{}, error) {
+		if url := req.URL(); url != "" {
+			return relationInfo{
+				URL: req.URL(),
+			}, nil
+		}
+		return nil, httpservice.ErrRelationIncomplete
+	}
+}
+
+func setFieldValue(f **mgo.Session, info relationInfo) error {
+	if *f != nil {
+		return httpservice.ErrRestartNeeded
+	}
+	session, err := mgo.Dial(info.URL)
```

Comment:
What's the decision behind Dialing here? You'd expect most handlers to dial as the first thing they do anyway. And setting up a relation has nothing really to do with "knowing" there is a mongo there. Doesn't this kind of assume implementation in the handler?